Python/ceval.c

```diff
@@ -761,12 +761,19 @@ _PyObjectArray_Free(PyObject **array, PyObject **scratch)
  * so consume 3 units of C stack */
 #define PY_EVAL_C_STACK_UNITS 2
 
-#if defined(_MSC_VER) && defined(_Py_USING_PGO) && defined(_Py_JIT)
-/* _PyEval_EvalFrameDefault is too large to optimize for speed with
-   PGO on MSVC when the JIT is enabled. Disable that optimization
-   around this function only. If this is fixed upstream, we should
-   gate this on the version of MSVC.
+
+/* _PyEval_EvalFrameDefault is too large to optimize for speed with PGO on MSVC
+   when the JIT is enabled or GIL is disabled. Disable that optimization around
+   this function only. If this is fixed upstream, we should gate this on the
+   version of MSVC.
  */
+#define DO_NOT_OPTIMIZE_INTERP_LOOP ( \
+    (defined(_MSC_VER) && \
+     defined(_Py_USING_PGO) && \
+     (defined(_Py_JIT) || \
+      defined(Py_GIL_DISABLED))))
```

Comment:
Looks like this style of define may not be portable (see warnings below). In other places, we do something like:

```
#if (defined(_MSC_VER) && \
     defined(_Py_USING_PGO) && \
     (defined(_Py_JIT) || \
      defined(Py_GIL_DISABLED)))
#define DO_NOT_OPTIMIZE_INTERP_LOOP
#endif
```

Reply:
Makes sense.  Done.